### DISCUSSION TITLE
8266 Modbus

### DISCUSSION THREAD

**Discussion by DC-555** [arendst/Tasmota#19157] [Q&A] [answered]
Good morning, I would like to put my 8266 to work with Modbus.
I already have a working Esp32.
But in 8266 it does not receive the request, it just sends it.
OBS- If I put the RX in esp32 and the TX in the 8266, I can receive the response from the modbus.
Template image:
![image](https://github.com/arendst/Tasmota/assets/128267457/641881b6-6897-4eaa-810e-cb349af4308c)

Attempting to get the response in the console:

`10:10:06.452 CMD: ModbusSerialConfig 3`
`10:10:06.460 MQT: stat/tasmota_3EDB6D/RESULT = {"ModbusSerialConfig":"8N1"}`
`10:10:21.934 CMD: ModbusBaudRate`
`10:10:21.942 MQT: stat/tasmota_3EDB6D/RESULT = {"ModbusBaudrate":9600}`
`10:10:26.986 CMD:  ModBusSend {"deviceaddress": 1, "functioncode": 3, "startaddress": 0, "type":"uint16", "count":1}`
`10:10:26.996 MQT: stat/tasmota_3EDB6D/RESULT = {"ModbusSend":"Done"}`

status0:
`00:00:02.147 RSL: STATUS0 = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_3EDB6D","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0},"StatusPRM":{"Baudrate":9600,"SerialConfig":"8N1","GroupTopic":"axon_connect_v1","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Exception","Uptime":"0T00:00:03","StartupUTC":"","Sleep":50,"CfgHolder":4617,"BootCount":21,"BCResetTime":"2023-07-19T17:04:47","SaveCount":39,"SaveAddress":"1000"},"StatusFWR":{"Version":"13.0.0.2(tasmota)","BuildDateTime":"2023-07-19T16:53:39","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"379/699"},"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Connect",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004001","00000000"]},"StatusMEM":{"ProgramSize":791,"Free":1256,"Heap":13,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640D8","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","879AD78F","043E8001","000080CF","010013C0","C000F983","00004084","00201000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,38,45,50,62,63,68","Sensors":"1,2,3,4,5,6,12,51","I2CDriver":"7,13"},"StatusNET":{"Hostname":"tasmota-3EDB6D-7021","IPAddress":"192.168.97.132","Gateway":"192.168.100.4","Subnetmask":"255.255.240.0","DNSServer1":"192.168.100.45","DNSServer2":"192.168.100.49","Mac":"40:F5:20:3E:DB:6D","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0},"StatusMQT":{"MqttHost":"staging.connect.axonlab.com.br","MqttPort":8883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_3EDB6D","MqttUser":"axontech","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4},"StatusTIM":{"UTC":"1970-01-01T00:00:02","Local":"1970-01-01T00:00:02","StartDST":"2023-10-29T02:00:00","EndDST":"2023-02-26T03:00:00","Timezone":99,"Sunrise":"21:43","Sunset":"10:48"},"StatusSNS":{"Time":"1970-01-01T00:00:02"},"StatusSTS":{"Time":"1970-01-01T00:00:02","Uptime":"0T00:00:03","UptimeSec":3,"Heap":12,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"Connect","BSSId":"06:EC:DA:FD:0E:08","Channel":6,"Mode":"11n","RSSI":66,"Signal":-67,"LinkCount":1,"Downtime":"0T00:00:03"}},"StatusSTK":{"Exception":29,"Reason":"Exception","EPC":["4026ec69","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000","CallChain":["40253bcd","401012d0","40253f09","40253f8e","40254068","402327dc","4023281e","4023435d","40229168","4023f972","40263200","402298d7","40247b9c","40229168","40269c28","40269bca","40269c28","401000e1","402291a0","40262d15","40231a6a","40101200","40259e35","40231b83","40231d56","40231da9","401012d0","40225cb0","40226e8b","402500ac","40250158"]}}
`

user_config_override.h :
`#ifndef USE_SERIAL_BRIDGE`
`#define USE_SERIAL_BRIDGE`
`#endif `
`#ifndef USE_MODBUS_BRIDGE`
`#define USE_MODBUS_BRIDGE`
`#endif`

The expected response is something like this, but it never arrives:
{"ModbusReceived":{"DeviceAddress":1,"FunctionCode":3,"StartAddress":0,"Length":7,"Count":1,"Values":[715]}}

Another thing observed is that the RX pin blinks only when RX is not connected to the 8266 board, I tested it on 3 different boards and got the same error.

**Comment by pkkrusty** [comment, marked as answer]
I am using Modbus Bridge successfully with multiple ESP8266. Are you feeding the RS485 board with 5v?

**Reply by DC-555** [reply]
Good morning.
Yes I am using it, the card I use is the same as the example, and it works correctly in esp32.
What version of Tasmota are you using?

**Reply by DC-555** [reply]
I did it, with your tip I went to test the power supply and put it in the same as the slave and it worked correctly, I believe I will have to make a filter.
Thanks.

**Reply by pkkrusty** [reply]
Yes, all my ESP8266 are using power supply from the slave, so common ground.
